device/tt_silicon_driver.cpp

```diff
@@ -48,6 +48,10 @@
 #include "tt_device.h"
 #include "ioctl.h"
 
+#include "src/firmware/riscv/grayskull/host_mem_address_map.h"
+#include "src/firmware/riscv/wormhole/host_mem_address_map.h"
+#include "src/firmware/riscv/blackhole/host_mem_address_map.h"
```

Comment:
We should really move all of this to TTDevice/architecture_implementation

Reply:
Move the include files you mean?

Reply:
I meant, architecture_implementation should hold something like "get_host_address_params", then each specialization per arch should return it's constants.
The code in tt_SiliconDevice would then be just calling AnyPCIDevice->get_architecture_implementation()->get_host_address_params.

Then you don't need these includes here

Reply:
I took your advice, and ran into a nasty circular header file dependency lol!

However, with the fix, I guess the code base is now better than ever!